### ISSUE TITLE
Compiler Support

### ISSUE THREAD

**Issue by rosenbrockc** [rosenbrockc/fortpy#33]
We need fortpy to handle multiple compilers and versions. The input/output files also need to vary based on the type of compiler (because they handle the random seeds differently). Here is a proposal:

Create a file called `compilers.xml`, it should have a list of compilers that fortpy can use to test with:

```xml
<compiler path="/usr/bin/gfortran_4.9.0" suffix="g49" name="gfortran49" default="true" />
```

- **_path_** specifies the full path to the compiler executable that fortpy should use.
- **_suffix_** if specified, this suffix will be spliced into _all_ input/output files that have `[c]` in the filename (i.e. `[c]` will be replaced by this suffix before copying input/output files to execution directories).
- **_name_** for the `-compiler` argument of `runtests.py`. Change that argument to accept a string. If the string is `*`, then run the unit tests separately _for every compiler_ in the `compilers.xml` file. Otherwise, only run those compilers whose names match the string given.
- **_default_** if `-compiler` is run _without_ specifying a string, this is the compiler that would be used by default. If none of the compilers in the file have this attribute, the first one listed is used.

The unit tests would be configured to create directories like `bcs.do_reweighted.g49`, one for each compiler with its suffix. The contents would be exclusive to that compiler then. Ideally, unit tests should be run for every minor compiler version.

**Comment by glwhart** [comment]
Sounds like a good solution.

**Comment by rosenbrockc** [comment]
Available starting in Revision 1.5.4
